### PR TITLE
fix(compass-sidebar): convert to ts, fix save connections bug COMPASS-7418

### PR DESCRIPTION
**Pull request by addaleax**
### Checklist
- [ ] New tests and/or benchmarks are included
- [ ] Documentation is changed or added
- [ ] I have signed the MongoDB Contributor License Agreement (https://www.mongodb.com/legal/contributor-agreement)

## Motivation and Context

- [x] Bugfix
- [ ] New feature
- [ ] Dependency update
- [x] Misc

## Types of changes

- [ ] *Backport Needed*
- [x] Patch (non-breaking change which fixes an issue)
- [ ] Minor (non-breaking change which adds functionality)
- [ ] Major (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: packages/compass-sidebar/src/modules/connection-info.ts [hunk 1]
```diff
@@ -25,11 +40,23 @@ export const INITIAL_STATE = {
...
+  | ChangeConnectionInfoAction
+  | SaveConnectionInfoAction;
+
+async function saveConnectionInfo(
+  connectionInfo: ConnectionInfo,
+  connectionStorage: typeof ConnectionStorage
+) {
   try {
-    await connectionStorage.save(connectionInfo);
+    await connectionStorage.save({ connectionInfo });
```
Comment (addaleax) [inline packages/compass-sidebar/src/modules/connection-info.ts:57]: ^ This is what was preventing connections from being saved

File: packages/compass-sidebar/src/modules/collection.js [hunk 2]
```diff
@@ -1,48 +0,0 @@
-import toNS from 'mongodb-ns';
```
Comment (addaleax) [inline packages/compass-sidebar/src/modules/collection.js:1]: This file was just completely unused, so I just removed it

File: packages/compass-sidebar/test/setup.js [hunk 3]
```diff
@@ -1,15 +0,0 @@
-const hadronApp = require('hadron-app');
```
Comment (addaleax) [inline packages/compass-sidebar/test/setup.js:1]: This file and the ones in test/mocks as well, they didn’t seem to actually be consumed anywhere so I just removed them

File: packages/compass-sidebar/src/modules/connection-info.ts (hunk 1 above)
Comment (mcasimir) [reply packages/compass-sidebar/src/modules/connection-info.ts:57]: Like completely?

**Review by mcasimir — APPROVED** [review]

File: packages/compass-sidebar/src/modules/connection-info.ts (hunk 1 above)
Comment (addaleax) [reply packages/compass-sidebar/src/modules/connection-info.ts:57]: Yeah, we don’t have an e2e test for this

**Pull request merged**